src/components/common/atoms/Input.tsx

```diff
@@ -1,32 +1,64 @@
-import Button from "./Button";
-import { InputModule } from "./inputStyle";
+import { ChangeEvent, ReactNode, useEffect, useRef, useState } from 'react';
+import Button from './Button';
+import { InputModule } from './inputStyle';
 interface IButtonModule {
   $type?: string;
   $inputClass?: string;
   $btnShow?: boolean;
-  $btnText?: string;
   $placeholder?: string;
   $beforeBgIcon?: string;
   $btnClass?: string;
+  children?: ReactNode;
+  $clickEvent?: string;
+  onchange?: (value: string) => void;
 }
+
 function Input({
   $btnShow = false,
-  $type = "text",
+  $type = 'text',
   $inputClass,
   $placeholder,
-  $btnText,
-  $beforeBgIcon = "",
-  $btnClass = "",
+  $beforeBgIcon = '',
+  $btnClass = '',
+  children,
+  $clickEvent,
+  onchange,
 }: IButtonModule) {
+  const [value, setValue] = useState('');
```

Comment:
```suggestion
  const [value, setValue] = useState<string>('');
```
string 타입같은 경우엔 확실히 타입을 정해주시는 게 좋아요!